addons/website_sale_collect/models/payment_transaction.py

```diff
@@ -0,0 +1,19 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    def _post_process(self):
+        """ Override of `payment` to confirm orders with on_site payment method to trigger a picking
+         creation.
+        """
+        on_site_pending_txs = self.filtered(
+            lambda tx: tx.provider_id.custom_mode == 'on_site' and tx.state == 'pending'
+        )
+        super(PaymentTransaction, self - on_site_pending_txs)._post_process()
```

Comment:
Shouldn't we also call super for `on_site_pending_txs`? Otherwise they won't be marked as post-processed I believe.

Reply:
I thought that overrides in other modules trigger unnecessary behavior, but after looking at it again, I see that it is indeed needed. Thank you for that catch!